### PR TITLE
Adding getting started guide

### PR DESCRIPTION
**Pull request by sshniro**
The following PR contains a getting started guide to quickly onboard new users to APISIX
The document contains the following steps:
1. Setting up APISIX
2. General Principles
3. Setup a route & upstream
4. Protect the API with an API key

Fix for: #930

### CODE REVIEW

File: doc/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+- Port: 443
+- URI: get
+- Query Parameters: foo1, foo2
+
+## Prerequisites
+
+- The guide uses docker and docker-compose to setup APISIX. But if you have already installed APISIX via other methods, then you can skip to step 2.
+- Curl: The guide uses curl command for API testing, but you can also use any other tool of your choice (Eg- Postman).
+
+## Step 1: Install API SIX
```
Comment (moonming) [inline doc/getting-started.md]: typo: `API SIX` -> `APISIX`

File: doc/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+APISIX is available to install in multiple operating environments. The [following document](how-to-build-cn.md#installation-via-source-release) shows the installation steps in multiple platforms.
```
Comment (moonming) [inline doc/getting-started.md]: `how-to-build-cn.md` -> `how-to-build.md`

File: doc/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+By default APISIX proxies the request via the HTTP protocol. As our backend is hosted in a HTTPS environment, let's use the proxy-rewrite plugin to change the scheme to HTTPS.
+
+```bash
+curl "http://127.0.0.1:9080/apisix/admin/routes/5" -X PUT -d '
+{
+    "uri": "/get",
+    "host": "httpbin.org",
+    "plugins": {
+                "proxy-rewrite": {
```
Comment (membphis) [inline doc/getting-started.md]: bad indentation

File: doc/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+
+Now, let's configure our endpoint to include the key-auth plugin.
+
+```bash
+curl http://127.0.0.1:9080/apisix/admin/routes/5 -X PUT -d '
+{
+    "uri": "/get",
+    "host": "httpbin.org",
+    "plugins": {
+                "proxy-rewrite": {
```
Comment (membphis) [inline doc/getting-started.md]: bad indentation

**Comment by membphis** [comment]
Please add a link of `doc/getting-started.md` to `doc/README.md`.

**Comment by sshniro** [comment]
> Please add a link of `doc/getting-started.md` to `doc/README.md`.

@membphis I have added the link to the ReadMe file.

**Comment by moonming** [comment]
merged. thx for your great job @sshniro

**Pull request merged**